examples/sugarscape_g1mt/sugarscape_g1mt/trader_agents.py

```diff
@@ -50,48 +50,13 @@ def __init__(
         self.prices = []
         self.trade_partners = []
 
-    def get_sugar(self, pos):
-        """
-        used in self.get_sugar_amount()
-        """
-
+    def get_resource(self, pos):
         this_cell = self.model.grid.get_cell_list_contents(pos)
         for agent in this_cell:
-            if type(agent) is Sugar:
+            if type(agent) is Resource:
                 return agent
         return None
```

Comment:
I don't think this ever called. Under this set up there is a resource agent in every cell even if they have 0 sugar and 0 spice. If it was called line 260 and 261 would error out with `None + int` Correct?

Reply:
I agree that the `return None` not being called because every cells has a `Resource` agent is rather an implicit consequence. A type checker would have said that `self.get_resource(pos).sugar_amount` does not hold for all cases.

It should be replaced with a patch/cell object soon, so that the return value of `get_resource` is `Resource`, not `Resource | None`.

Reply:
I raised an `Exception` in `get_resource` when no `Resource` agent is found. This should pass Mypy check.